packages/expo-permissions/ios/EXPermissions/EXUserNotificationRequester.m

```diff
@@ -0,0 +1,102 @@
+// Copyright © 2018 650 Industries. All rights reserved.
+
+#import <EXPermissions/EXUserNotificationRequester.h>
+
+#import <UIKit/UIKit.h>
+#import <EXPermissionUserNotificationCenter.h>
+
+@interface EXUserNotificationRequester ()
+
+@property (nonatomic, strong) EXPromiseResolveBlock resolve;
+@property (nonatomic, strong) EXPromiseRejectBlock reject;
+@property (nonatomic, weak) id<EXPermissionRequesterDelegate> delegate;
+@property (nonatomic, weak) EXModuleRegistry * moduleRegistry;
+
+@end
+
+@implementation EXUserNotificationRequester
+
+- (instancetype)initWithModuleRegistry: (EXModuleRegistry *) moduleRegistry {
+  if (self = [super init]) {
+    _moduleRegistry = moduleRegistry;
+  }
+  return self;
+}
+
++ (id<EXUserNotificationsPermissionsCenterInterface>) getCenterWithModuleRegistry:(EXModuleRegistry *) moduleRegistry {
+  return [moduleRegistry getModuleImplementingProtocol:@protocol(EXUserNotificationsPermissionsCenterInterface)];
+}
+
++ (NSDictionary *)permissionsWithModuleRegistry:(EXModuleRegistry *)moduleRegistry
+{
+  dispatch_semaphore_t sem = dispatch_semaphore_create(0);
+  __block BOOL allowsSound;
+  __block BOOL allowsAlert;
+  __block BOOL allowsBadge;
+  __block EXPermissionStatus status;
+
+  [[EXUserNotificationRequester getCenterWithModuleRegistry:moduleRegistry] getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings *settings) {
+    allowsSound = settings.soundSetting;
+    allowsAlert = settings.alertStyle;
+    allowsBadge = settings.badgeSetting;
+
+    status = EXPermissionStatusUndetermined;
+
+    if (settings.authorizationStatus == UNAuthorizationStatusAuthorized) {
```

Comment:
I believe we can treat Provisional as if the permission were granted, but need to test

Reply:
When we get "provisional" permission (this means being able to send a notification without showing the permission dialog to the user), we'll get back UNAuthorizationStatusProvisional and should send back "granted" (and maybe `"provisional": true` in the dictionary).